pkg/skaffold/schema/defaults/defaults.go

```diff
@@ -217,8 +219,8 @@ func setDefaultClusterDockerConfigSecret(cluster *latest.ClusterDetails) error {
 	if cluster.DockerConfig == nil {
 		return nil
 	}
-
-	cluster.DockerConfig.SecretName = valueOrDefault(cluster.DockerConfig.SecretName, constants.DefaultKanikoDockerConfigSecretName)
+	uuid, err := uuid.NewUUID()
```

Comment:
This should be guarded by a flag and not happen everytime.

Reply:
you mean

```
if cluster.DockerConfig.SecretName== nil {
       uuid, err := uuid.NewUUID()
       cluster.DockerConfig.SecretName = valueOrDefault(cluster.DockerConfig.SecretName, constants.DefaultKanikoDockerConfigSecretName+uuid.String())		
}
```

Reply:
Do you mean adding field in skaffold.yaml like
```
cluster:
  randomSecret: True
```

Reply:
yes! sounds good.